pkg/reconciler/openshift/tektonconfig/rbac.go

```diff
@@ -81,25 +83,84 @@ func (r *rbac) createResources(ctx context.Context) error {
 
 	logger := logging.FromContext(ctx)
 
-	// fetch the list of all namespaces which doesn't have label
-	// `openshift-pipelines.tekton.dev/namespace-ready: <release-version>`
-	namespaces, err := r.kubeClientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("%s != %s", namespaceVersionLabel, r.version),
-	})
+	// list of namespaces rbac resources need to be created
+	var rbacNamespaces []corev1.Namespace
+	// list of namespaces rbac resources need not to be created
+	var nonRbacNamespaces []corev1.Namespace
+
+	// list all namespaces
+	allNamespaces, err := r.kubeClientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
 
-	// list of namespaces rbac resources need to be created
-	var rbacNamespaces []corev1.Namespace
-
-	// filter namespaces:
-	// ignore ns with name passing regex `^(openshift|kube)-`
-	for _, n := range namespaces.Items {
+	for _, n := range allNamespaces.Items {
 		if ignore := nsRegex.MatchString(n.GetName()); ignore {
 			continue
 		}
-		rbacNamespaces = append(rbacNamespaces, n)
+		v, ok := n.GetLabels()[disableRbacInjection]
+		if r.disableRbacClusterWide {
+			if ok {
+				if v == "false" || v == "" {
+					rbacNamespaces = append(rbacNamespaces, n)
```

Comment:
previously, after creating the rbac resources in a `ns` we use to add `openshift-pipelines.tekton.dev/namespace-reconcile-version = <version>` label and in next reconcilation if ns has label then we use to ignore it
unless if it is an upgrade then the check will fail and we recreate rbac 

Currently, it is going through all ns each time right? do we not need to check the label
if the label exist then we can ignore going though that ns for rbac creation

Reply:
>and in next reconcilation if ns has label then we use to ignore it
unless if it is an upgrade then the check will fail and we recreate rbac

right but for this scenario everytime user update namespace label and in that case if we list namespace based on `openshift-pipelines.tekton.dev/namespace-reconcile-version = <version>` then while updating label on namespace it was never coming to this logic

Reply:
i think if we split the namespace list query based on teh clusterWide spec, the list of namespaces will be much shorter.
This could fasten the process a lot as practically we will be interested in on only a short subset of all namespaces in any cluster. Let me try to come up with an example.